lib/input/sftp.go

```diff
@@ -108,21 +140,43 @@ type sftpReader struct {
 	scannerMut  sync.Mutex
 	scanner     codec.Reader
 	currentPath string
+
+	watcherPollInterval time.Duration
+	watcherCache        types.Cache
 }
 
-func newSFTPReader(conf SFTPConfig, log log.Modular, stats metrics.Type) (*sftpReader, error) {
+func newSFTPReader(conf SFTPConfig, mgr types.Manager, log log.Modular, stats metrics.Type) (*sftpReader, error) {
 	codecConf := codec.NewReaderConfig()
 	codecConf.MaxScanTokenSize = conf.MaxBuffer
 	ctor, err := codec.GetReader(conf.Codec, codecConf)
 	if err != nil {
 		return nil, err
 	}
 
+	var watcherPollInterval time.Duration
+	var cache types.Cache
+	if conf.Watcher.Enabled {
+		if watcherPollInterval, err = time.ParseDuration(conf.Watcher.PollInterval); err != nil {
+			return nil, fmt.Errorf("failed to parse watcher poll interval string: %v", err)
+		}
+
+		if conf.Watcher.Cache == "" {
+			return nil, fmt.Errorf("watcher cache is required if watcher mode is enabled")
+		}
+
+		cache, err = mgr.GetCache(conf.Watcher.Cache)
```

Comment:
Resources right now are static but there's no guarantee that they won't change over time, so this should stick around as a check that the cache exists but do not capture it in a variable. Each time it gets referenced below you need to call `mgr.GetCache` again.

```suggestion
		_, err = mgr.GetCache(conf.Watcher.Cache)
```